charts/argo-workflows/templates/notify-slack-workflow.yaml

```diff
@@ -0,0 +1,37 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: notify-slack
+spec:
+  entrypoint: notify-slack
+  arguments:
+    parameters:
+      - name: slackChannel
+      - name: text
+  templates:
+    - name: notify-slack
+      inputs:
+        parameters:
+        - name: slackChannel
+        - name: text
+      container:
+        image: curlimages/curl
+        command:
+          - "curl"
+          - "-X"
+          - "POST"
+          - "--data-urlencode"
+          - >
+              payload={
+                "channel": "#{{"{{inputs.parameters.slackChannel}}"}}",
```

Comment:
optional: Might be worth a comment here about the double-templating - probably quite useful for the reader to know that it's Jinja inside Go template or whatever it is. Similarly for the one in post-sync-workflow.yaml.

Or could pull this out into a file too, but that's probably overkill here since having to look through a zillion files also detracts from readability 🤷

Reply:
done thanks